funflow/src/Control/Funflow.hs

```diff
@@ -24,16 +26,46 @@ module Control.Funflow
   , Class.stepIO
   , Class.wrap
   , CS.withStore
+  , hoistFlowEff
+  , expandFlowEff
   , module Control.Funflow.Steps
   , module Control.Funflow.Exec.Simple
   , module Control.Funflow.External
   )
   where
 
-import qualified Control.Funflow.Base         as Base
+import           Control.Arrow.Free
+import           Control.Funflow.Base         as Base
 import qualified Control.Funflow.Cache.TH     as Cache
 import qualified Control.Funflow.Class        as Class
 import qualified Control.Funflow.ContentStore as CS
 import           Control.Funflow.Exec.Simple
 import           Control.Funflow.External
 import           Control.Funflow.Steps
+
+
+-- | Change the effect type present inside the 'Flow'
+hoistFlowEff :: (forall x y. eff x y -> eff' x y) -> Flow eff ex a b -> Flow eff' ex a b
+hoistFlowEff f = hoistErrorChoiceEff $ \e -> case e of
+  Step p s -> Step p s
+  StepIO p s -> StepIO p s
+  External p s -> External p s
+  PutInStore s -> PutInStore s
+  GetFromStore s -> GetFromStore s
+  InternalManipulateStore s -> InternalManipulateStore s
+  Wrapped p s -> Wrapped p (f s)
+
+-- | Turns each effect in the 'Flow' as a 'Flow' of some other effect
```

Comment:
This comment is pretty cryptic to me! Would it be possible to furnish this with an example of its usage?